connector/datadogconnector/factory.go

```diff
@@ -38,10 +47,15 @@ func createDefaultConfig() component.Config {
 
 // defines the consumer type of the connector
 // we want to consume traces and export metrics therefore define nextConsumer as metrics, consumer is the next component in the pipeline
-func createTracesToMetricsConnector(_ context.Context, params connector.Settings, cfg component.Config, nextConsumer consumer.Metrics) (connector.Traces, error) {
+func createTracesToMetricsConnector(_ context.Context, params connector.Settings, cfg component.Config, nextConsumer consumer.Metrics) (c connector.Traces, err error) {
 	metricsClient := metricsclient.InitializeMetricClient(params.MeterProvider, metricsclient.ConnectorSourceTag)
-	timingReporter := timing.New(metricsClient)
-	c, err := newTraceToMetricConnector(params.TelemetrySettings, cfg, nextConsumer, metricsClient, timingReporter)
+	if NativeIngestFeatureGate.IsEnabled() {
+		params.Logger.Info("Datadog connector using the native OTel API to ingest OTel spans and produce APM stats")
+		c, err = newTraceToMetricConnectorNative(params.TelemetrySettings, cfg, nextConsumer, metricsClient)
+	} else {
+		params.Logger.Info("Datadog connector using the legacy processing pipelines to ingest OTel spans and produce APM stats")
```

Comment:
Should this log mention the feature gate to nudge people to opt in into the feature? As it stands it can be a bit confusing ("legacy" implies "not recommended" to me, but I don't have any actionable stuff to do when I see this log message)

Reply:
Good point, I added the feature gate in this message